src/FakeItEasy/Creation/CreationResult.cs

```diff
@@ -46,40 +46,29 @@ private class UntriedCreationResult : CreationResult
             public override CreationResult MergeIntoDummyResult(CreationResult other) => other;
         }
 
-        private class SuccessfulCreationResult : CreationResult
+        private class SuccessfulCreationResult(object? result) : CreationResult

```

Comment:
You have some reservations about primary constructors that I forget and didn't understand at the time. I can drop such changes if you like (and configure Rider not to suggest them to me in the future).

Reply:
My reservation is mostly that almost every time I try to use them, I end up needing additional constructor logic, so I need to create an explicit constructor... But it's not the case here.